### DISCUSSION TITLE
nvim-java organize import like  nvim-jdtls?

### DISCUSSION THREAD

**Discussion by chetra-seng** [nvim-java/nvim-java#254] [Q&A] [answered]
I'm currently using Lazyvim with nvim-java. However, it seems to miss the code organize imports feature. If I use Lazyvim extras, which automatically configure nvim-jdtls, I can get the organize imports feature. It didn't affect nvim-java configurations and all the features also worked fine. The only problem I have is, whenever the language server starts initializing, I get this warning: 
```bash
Client 1 quit with exit code 1 and signal 0
```
Is there a way to fix this?
Edit: I am also aware that nvim-jdtls is not compatible with nvim-java, I just want to have code organize imports feature working in nvim-java :)

**Comment by s1n7ax** [comment]
I've been working on refactoring commands. There are quite a lot to be done. But eventually all the client commands applicable to neovim will be added. You can't use jdtls and nvim-java both at the same time. Can't help you with any of the issues you are facing when both used. For now, you can use this.
https://github.com/orgs/nvim-java/discussions/200#discussioncomment-9611677

**Reply by chetra-seng** [reply]
Thanks @s1n7ax. I would like to ask you for further elaboration. Based on this configuration from wiki for Lazyvim
```lua
return {
  'nvim-java/nvim-java',
  config = false,
  dependencies = {
    {
      'neovim/nvim-lspconfig',
      opts = {
        servers = {
          jdtls = {
            -- Configs for organize imports here ? 
          },
        },
        setup = {
          jdtls = function()
            require('java').setup({
              -- Your custom nvim-java configuration goes here
            })
          end,
        },
      },
    },
  },
}
```
I just need to include the configuration into jdtls section, is that correct?

**Comment by s1n7ax** [comment, marked as answer]
@chetra-seng No. Paste the entire snippet in `init.lua` or something. `nvim_create_autocmd` add something to be ran on an event. `BufWritePre` event is triggerd before buffer writes to file. In the callback, it runs organize import based on the file type.

**Comment by s1n7ax** [comment]
@chetra-seng From what I can see, lazyvim have keymap to run the organize import but it does not organize imports on save. Do you mean organize import on keymap?

**Reply by chetra-seng** [reply]
Yes, I was referring to organizing imports keymap

**Reply by chetra-seng** [reply]
It would be a great if you could help with that.

**Reply by s1n7ax** [reply]
@chetra-seng You have to learn neovim basics eventually. Give it a try by yourself.

Following will format the code. `:h vim.keymap` for more information about keymaps

```lua
vim.lsp.buf.code_action({
  context = { only = { "source.organizeImports" } },
  apply = true,
})
```

**Reply by chetra-seng** [reply]
@s1n7ax Huge thanks for all the contributions and supports. This might be out of context, however, it would greatly help me if you could share some good resources to learn Neovim basics. Thanks :)))

**Reply by s1n7ax** [reply]
https://github.com/nvim-java/nvim-java/wiki/Tips-&-Tricks#running-code-actions
